go/ct/gen/transaction_context_test.go

```diff
@@ -33,3 +94,205 @@ func TestTransactionContext_Generate(t *testing.T) {
 		t.Errorf("Generated origin address has default value.")
 	}
 }
+
+func TestTransactionContextGenerator_GenerateConstrained(t *testing.T) {
+
+	variableCheck := func(txCtx st.TransactionContext, assignment Assignment, t *testing.T,
+		variable Variable, shouldBePresent, shouldBeAssigned bool, value common.U256) {
+		t.Helper()
+		assignedValue, ok := assignment[variable]
+		if !ok {
+			t.Errorf("Variable %v should be in assignment.", variable.String())
+		}
+		if shouldBePresent {
+			if !assignedValue.IsUint64() {
+				t.Errorf("Variable %v should be assigned a uint64 value.", variable.String())
+			}
+			if assignedValue.Uint64() >= uint64(len(txCtx.BlobHashes)) {
+				t.Errorf("Assigned value for %v %v is out of range.", variable.String(), assignedValue.Uint64())
+			}
+		} else {
+			if assignedValue.Uint64() < uint64(len(txCtx.BlobHashes)) {
+				t.Errorf("Assigned value for %v %v is not out of range.", variable.String(), assignedValue.Uint64())
+			}
+		}
+		if shouldBeAssigned {
+			if !assignedValue.Eq(value) {
+				t.Errorf("Assigned value for %v %vis not the expected value.", variable.String(), assignedValue.Uint64())
+			}
+		}
+	}
+
+	tests := map[string]struct {
+		setup func(*TransactionContextGenerator, *Assignment)
+		check func(st.TransactionContext, Assignment, *testing.T)
+	}{
+		"present": {
+			setup: func(txCtxGen *TransactionContextGenerator, _ *Assignment) {
+				txCtxGen.PresentBlobHashIndex(Variable("v1"))
+			},
+			check: func(txCtx st.TransactionContext, assignment Assignment, t *testing.T) {
+				variableCheck(txCtx, assignment, t, Variable("v1"), true, false, common.NewU256(0))
```

Comment:
If you follow the suggestion listed above, what remains of the "variableCheck" is a list of variables that are in our outside the branch range. In this case, it would be more readable to redefine the test struct as
```
struct {
   setup func(*TransactionContextGenerator, *Assignment)
   shouldBePresent []Variable
   shouldBeAbsend []Variable
```
and then inline the `variableCheck` function in the test loop below.